crates/fuel-core-nats/src/lib.rs

```diff
@@ -138,89 +272,78 @@ pub async fn nats_publisher(
             // Publish the transaction.
             info!("NATS Publisher: Transaction#{height}.{index}.{tx_kind}");
             let payload = serde_json::to_string_pretty(tx)?;
-            jetstream
-                .publish(
-                    format!("transactions.{height}.{index}.{tx_kind}"),
-                    payload.into(),
-                )
-                .await?;
+            self.publish(
+                format!("transactions.{height}.{index}.{tx_kind}"),
+                payload.into(),
+            )
+            .await?;
```

Comment:
Would it be cleaner to extract `publish_transaction_kind` into a separate function?